pkg/ucp/api/v20231001preview/aws_credential_conversion.go

```diff
@@ -82,13 +82,47 @@ func (cr *AwsCredentialResource) getDataModelCredentialProperties() (*datamodel.
 		}
 
 		return &datamodel.AWSCredentialResourceProperties{
-			Kind: datamodel.AWSCredentialKind,
+			Kind: datamodel.AWSAccessKeyCredentialKind,
 			AWSCredential: &datamodel.AWSCredentialProperties{
-				AccessKeyID:     to.String(p.AccessKeyID),
-				SecretAccessKey: to.String(p.SecretAccessKey),
+				Kind: datamodel.AWSAccessKeyCredentialKind,
+				AccessKeyCredential: &datamodel.AWSAccessKeyCredentialProperties{
+					AccessKeyID:     to.String(p.AccessKeyID),
+					SecretAccessKey: to.String(p.SecretAccessKey),
+				},
 			},
 			Storage: storage,
 		}, nil
+	case *AwsIRSACredentialProperties:
+		var storage *datamodel.CredentialStorageProperties
+
+		switch c := p.Storage.(type) {
+		case *InternalCredentialStorageProperties:
+			if c.Kind == nil {
+				return nil, &v1.ErrModelConversion{PropertyName: "$.properties", ValidValue: "not nil"}
```

Comment:
What's the right property path for this? It should be longer.

Reply:
Updated it to .properties.storage.kind